python/cudf/cudf/core/index.py

```diff
@@ -1443,7 +1443,21 @@ def __repr__(self):
                     output[:break_idx].replace("'", "") + output[break_idx:]
                 )
             else:
-                output = repr(preprocess.to_pandas())
+                # Too many non-unique categories will cause
+                # the output to take too long. In this case, we
+                # split the categories into data and categories
+                # and generate the repr separately and
+                # merge them.
+                pd_cats = pd.Categorical(
```

Comment:
I think you'll also need to pass `ordered=preprocess.ordered` here e.g. from this patch

```python
In [1]: import cudf

In [2]: ci = cudf.CategoricalIndex(range(1), ordered=True)

In [3]: ci
Out[3]: CategoricalIndex([0], categories=[0], ordered=False, dtype='category')
```

Reply:
oops, opening a fix

Reply:
Opened a fix: https://github.com/rapidsai/cudf/pull/16683